File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class DailyRevenue(BaseModel):
    date: str
    revenue: float


class TopProduct(BaseModel):
    name: str
    revenue: float


class OrderStatusCount(BaseModel):
    status: str
    count: int


class AnalyticsSummary(BaseModel):
    total_revenue: float
    total_orders: int
    completed_orders: int
    avg_order_value: float
    total_customers: int
    total_products: int
    low_stock_count: int
    out_of_stock_count: int
    revenue_by_day: list[DailyRevenue]
    top_products: list[TopProduct]
    orders_by_status: list[OrderStatusCount]

File: frontend/src/types/analytics.ts
export interface DailyRevenue {
  date: string;
  revenue: number;
}

export interface TopProduct {
  name: string;
  revenue: number;
}

export interface OrderStatusCount {
  status: string;
  count: number;
}

export interface AnalyticsSummary {
  total_revenue: number;
  total_orders: number;
  completed_orders: number;
  avg_order_value: number;
  total_customers: number;
  total_products: number;
  low_stock_count: number;
  out_of_stock_count: number;
  revenue_by_day: DailyRevenue[];
  top_products: TopProduct[];
  orders_by_status: OrderStatusCount[];
}
